packages/api-docs-builder/buildApiUtils.ts

```diff
@@ -357,6 +357,73 @@ export const getBaseHookInfo = (filename: string): HookInfo => {
   return result;
 };
 
+function findJoyUIDemos(
+  componentName: string,
+  pagesMarkdown: ReadonlyArray<{ pathname: string; title: string; components: readonly string[] }>,
+) {
+  return pagesMarkdown
+    .filter(
+      (page) => page.pathname.indexOf('/joy/') === 0 && page.components.includes(componentName),
+    )
+    .map((page) => ({
+      name: page.title,
+      demoPathname: replaceComponentLinks(`${page.pathname.replace(/^\/joy/, '')}/`).replace(
+        'material-ui',
+        'joy-ui',
+      ),
+    }));
+}
+
+export const getJoyComponentInfo = (filename: string): ComponentInfo => {
+  const { name } = extractPackageFile(filename);
+  let srcInfo: null | ReturnType<ComponentInfo['readFile']> = null;
+  if (!name) {
+    throw new Error(`Could not find the component name from: ${filename}`);
+  }
+  return {
+    filename,
+    name,
+    muiName: getMuiName(name),
+    apiPathname: `/joy-ui/api/${kebabCase(name)}/`,
+    apiPagesDirectory: path.join(process.cwd(), `docs/pages/joy-ui/api`),
+    isSystemComponent: systemComponents.includes(name),
+    readFile() {
+      srcInfo = parseFile(filename);
+      return srcInfo;
+    },
+    getInheritance(inheritedComponent = srcInfo?.inheritedComponent) {
+      if (!inheritedComponent) {
+        return null;
+      }
+      return {
+        name: inheritedComponent,
+        apiPathname:
+          inheritedComponent === 'Transition'
```

Comment:
We don't have `Transition` component in Joy UI, we can likely drop this condition.